### PR TITLE
Correct spelling

### PR DESCRIPTION
**Pull request by bf4**
```
go get -u github.com/client9/misspell/cmd/misspell
misspell  -w -error -source=text .
```

### CODE REVIEW

File: actionpack/test/controller/integration_test.rb [hunk 1]
```diff
@@ -494,7 +494,7 @@ def test_respect_removal_of_default_headers_by_a_controller_action
     assert_includes @response.headers, "c"
   end
 
-  def test_accept_not_overriden_when_xhr_true
+  def test_accept_not_overridden_when_xhr_true
```
Comment (bf4) [inline actionpack/test/controller/integration_test.rb]: `s/overriden/overridden`

File: actionpack/test/controller/mime/respond_to_test.rb [hunk 2]
```diff
@@ -519,7 +519,7 @@ def test_handle_any_any_xml
     assert_equal "Whatever you ask for, I got it", @response.body
   end
 
-  def test_handle_any_any_unkown_format
+  def test_handle_any_any_unknown_format
```
Comment (bf4) [inline actionpack/test/controller/mime/respond_to_test.rb]: `s/unkown/unknown`  also `any_any`, I think

File: actionpack/test/dispatch/static_test.rb [hunk 3]
```diff
@@ -163,7 +163,7 @@ def test_does_not_modify_path_info
     assert_equal file_name, env["PATH_INFO"]
   end
 
-  def test_serves_gzip_with_propper_content_type_fallback
+  def test_serves_gzip_with_proper_content_type_fallback
```
Comment (bf4) [inline actionpack/test/dispatch/static_test.rb]: `s/propper/proper`

File: actionview/test/ujs/public/vendor/jquery.metadata.js [hunk 4]
```diff
@@ -111,7 +111,7 @@ $.extend({
  *
  * @name metadata
  * @descr Returns element's metadata object
- * @param Object opts An object contianing settings to override the defaults
+ * @param Object opts An object containing settings to override the defaults
```
Comment (bf4) [inline actionview/test/ujs/public/vendor/jquery.metadata.js]: `s/contianing/containing`

File: activerecord/CHANGELOG.md [hunk 5]
```diff
@@ -171,7 +171,7 @@
     *Rafael Mendonça França*
 
 *   Set `:time` as a timezone aware type and remove deprecation when
-    `config.active_record.time_zone_aware_types` is not explictly set.
+    `config.active_record.time_zone_aware_types` is not explicitly set.
```
Comment (bf4) [inline activerecord/CHANGELOG.md]: `s/explictly/explicitly`

File: activerecord/test/cases/adapters/mysql2/active_schema_test.rb [hunk 6]
```diff
@@ -92,8 +92,8 @@ def (ActiveRecord::Base.connection).index_name_exists?(*); false; end
       assert_equal expected, actual
     end
 
-    expected = "ALTER TABLE `peaple` ADD  INDEX `index_peaple_on_last_name` USING btree (`last_name`(10)), ALGORITHM = COPY"
-    actual = ActiveRecord::Base.connection.change_table(:peaple, bulk: true) do |t|
+    expected = "ALTER TABLE `people` ADD  INDEX `index_people_on_last_name` USING btree (`last_name`(10)), ALGORITHM = COPY"
```
Comment (bf4) [inline activerecord/test/cases/adapters/mysql2/active_schema_test.rb]: `s/peaple/people`

File: activerecord/test/cases/adapters/mysql2/connection_test.rb [hunk 7]
```diff
@@ -120,7 +120,7 @@ def test_mysql_sql_mode_variable_overrides_strict_mode
     end
   end
 
-  def test_passing_arbitary_flags_to_adapter
+  def test_passing_arbitrary_flags_to_adapter
```
Comment (bf4) [inline activerecord/test/cases/adapters/mysql2/connection_test.rb]: `s/arbitary/arbitrary`

File: activerecord/test/cases/associations/has_and_belongs_to_many_associations_test.rb [hunk 8]
```diff
@@ -745,8 +745,8 @@ def test_find_scoped_grouped
   end
 
   def test_find_scoped_grouped_having
-    assert_equal 2, projects(:active_record).well_payed_salary_groups.to_a.size
-    assert projects(:active_record).well_payed_salary_groups.all? { |g| g.salary > 10000 }
+    assert_equal 2, projects(:active_record).well_paid_salary_groups.to_a.size
```
Comment (bf4) [inline activerecord/test/cases/associations/has_and_belongs_to_many_associations_test.rb]: `s/well_payed/well_paid`

File: activerecord/test/cases/associations/has_one_associations_test.rb [hunk 9]
```diff
@@ -661,15 +661,15 @@ class SpecialSupscription < ActiveRecord::Base
     belongs_to :book, class_name: "SpecialBook"
   end
 
-  def test_assocation_enum_works_properly
+  def test_association_enum_works_properly
```
Comment (bf4) [inline activerecord/test/cases/associations/has_one_associations_test.rb]: `s/assocation/association`

File: activerecord/test/cases/associations/has_one_associations_test.rb [hunk 10]
```diff
@@ -661,15 +661,15 @@ class SpecialSupscription < ActiveRecord::Base
...
     author = SpecialAuthor.create!(name: "Test")
     book = SpecialBook.create!(status: "published")
     author.book = book
 
     refute_equal 0, SpecialAuthor.joins(:book).where(books: { status: "published" }).count
   end
 
-  def test_assocation_enum_works_properly_with_nested_join
+  def test_association_enum_works_properly_with_nested_join
```
Comment (bf4) [inline activerecord/test/cases/associations/has_one_associations_test.rb]: `s/assocation/association`

File: activerecord/test/cases/connection_pool_test.rb [hunk 11]
```diff
@@ -441,7 +441,7 @@ def test_disconnect_and_clear_reloadable_connections_attempt_to_wait_for_threads
         end
       end
 
-      def test_bang_versions_of_disconnect_and_clear_reloadable_connections_if_unable_to_aquire_all_connections_proceed_anyway
+      def test_bang_versions_of_disconnect_and_clear_reloadable_connections_if_unable_to_acquire_all_connections_proceed_anyway
```
Comment (bf4) [inline activerecord/test/cases/connection_pool_test.rb]: `s/aquire/acquire`

File: activerecord/test/cases/finder_test.rb [hunk 12]
```diff
@@ -863,13 +863,13 @@ def test_bind_variables
   end
 
   def test_bind_variables_with_quotes
-    Company.create("name" => "37signals' go'es agains")
-    assert Company.where(["name = ?", "37signals' go'es agains"]).first
+    Company.create("name" => "37signals' go'es against")
```
Comment (bf4) [inline activerecord/test/cases/finder_test.rb]: `s/agains/against`

File: activerecord/test/fixtures/subscribers.yml [hunk 13]
```diff
@@ -6,6 +6,6 @@ second:
   nick: webster132
   name: David Heinemeier Hansson
 
-thrid:
+third:
```
Comment (bf4) [inline activerecord/test/fixtures/subscribers.yml]: `s/thrid/third`, but might be dead code

File: activerecord/test/models/project.rb [hunk 14]
```diff
@@ -11,7 +11,7 @@ class Project < ActiveRecord::Base
                             after_add: Proc.new { |o, r| o.developers_log << "after_adding#{r.id || '<new>'}" },
                             before_remove: Proc.new { |o, r| o.developers_log << "before_removing#{r.id}" },
                             after_remove: Proc.new { |o, r| o.developers_log << "after_removing#{r.id}" }
-  has_and_belongs_to_many :well_payed_salary_groups, -> { group("developers.salary").having("SUM(salary) > 10000").select("SUM(salary) as salary") }, class_name: "Developer"
+  has_and_belongs_to_many :well_paid_salary_groups, -> { group("developers.salary").having("SUM(salary) > 10000").select("SUM(salary) as salary") }, class_name: "Developer"
```
Comment (bf4) [inline activerecord/test/models/project.rb]: `s/well_payed/well_paid`

File: activesupport/test/autoloading_fixtures/raises_arbitrary_exception.rb [hunk 15]
```diff
@@ -1,4 +1,4 @@
 RaisesArbitraryException = 1
 _ = A::B # Autoloading recursion, also expected to be watched and discarded.
 
-raise Exception, "arbitray exception message"
+raise Exception, "arbitrary exception message"
```
Comment (bf4) [inline activesupport/test/autoloading_fixtures/raises_arbitrary_exception.rb]: `s/arbitrary/arbitrary`

File: actionpack/test/controller/filters_test.rb [hunk 16]
```diff
@@ -584,7 +584,7 @@ def test_running_actions_with_class
     assert @controller.instance_variable_get(:@was_audited)
   end
 
-  def test_running_anomolous_yet_valid_condition_actions
+  def test_running_anomalous_yet_valid_condition_actions
```
Comment (bf4) [inline actionpack/test/controller/filters_test.rb]: `s/anomolous/anomalous`

File: actionpack/lib/action_dispatch/journey/parser.rb [hunk 17]
```diff
@@ -1,7 +1,7 @@
 #
 # DO NOT MODIFY!!!!
 # This file is automatically generated by Racc 1.4.14
-# from Racc grammer file "".
+# from Racc grammar file "".
```
Comment (bf4) [inline actionpack/lib/action_dispatch/journey/parser.rb]: `s/grammer/grammar`

File: activesupport/test/core_ext/module/attribute_aliasing_test.rb [hunk 18]
```diff
@@ -52,8 +52,8 @@ def test_aliasing_to_uppercase_attributes
     assert_equal "No, really, this is not a joke.", e.Data
     assert e.Data?
 
-    e.Data = "Uppercased methods are teh suck"
-    assert_equal "Uppercased methods are teh suck", e.body
+    e.Data = "Uppercased methods are the suck"
```
Comment (bf4) [inline activesupport/test/core_ext/module/attribute_aliasing_test.rb]: `¯\_(ツ)_/¯`

File: activesupport/test/dependencies_test.rb [hunk 19]
```diff
@@ -272,7 +272,7 @@ def test_nested_class_can_access_sibling
   def test_raising_discards_autoloaded_constants
     with_autoloading_fixtures do
       e = assert_raises(Exception) { RaisesArbitraryException }
-      assert_equal("arbitray exception message", e.message)
+      assert_equal("arbitrary exception message", e.message)
```
Comment (bf4) [inline activesupport/test/dependencies_test.rb]: `s/arbitray/arbitrary`

File: activesupport/test/evented_file_update_checker_test.rb [hunk 20]
```diff
@@ -39,7 +39,7 @@ def touch(files)
     checker = new_checker(tmpfiles) {}
     assert !checker.updated?
 
-    # Pipes used for flow controll across fork.
+    # Pipes used for flow control across fork.
```
Comment (bf4) [inline activesupport/test/evented_file_update_checker_test.rb]: `s/controll/control`

File: activesupport/test/multibyte_chars_test.rb [hunk 21]
```diff
@@ -50,7 +50,7 @@ def test_methods_are_forwarded_to_wrapped_string_for_byte_strings
     assert_equal BYTE_STRING.length, BYTE_STRING.mb_chars.length
   end
 
-  def test_forwarded_method_with_non_string_result_should_be_returned_vertabim
+  def test_forwarded_method_with_non_string_result_should_be_returned_verbatim
```
Comment (bf4) [inline activesupport/test/multibyte_chars_test.rb]: `s/vertabim/verbatim`

File: railties/test/application/configuration_test.rb [hunk 22]
```diff
@@ -51,15 +51,15 @@ def app(env = "development")
 
     def setup
       build_app
-      supress_default_config
+      suppress_default_config
```
Comment (bf4) [inline railties/test/application/configuration_test.rb]: `s/supress/suppress`

**Review by bf4 — COMMENTED** [review]
Typos

**Comment by arthurnn** [comment]
thanks

File: actionpack/lib/action_dispatch/journey/parser.rb (hunk 17 above)
Comment (matthewd) [inline actionpack/lib/action_dispatch/journey/parser.rb]: 😒

**Comment by bf4** [comment]
@arthurnn :) any thoughts on https://github.com/rails/rails/pull/27919#discussion_r99506151 as indicating dead code?

File: activerecord/test/fixtures/subscribers.yml (hunk 13 above)
Comment (arthurnn) [inline activerecord/test/fixtures/subscribers.yml]: If nobody is using the fixture I think we could remove it. 
Didn't look much into it to be honest. But I am always for removing not used code.

**Comment by bf4** [comment]
Ok. I didn't want to expand the pr scope.  I'll look at making a followup

B mobile phone

**Pull request merged**